src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -4874,6 +4897,67 @@ int MethodContext::repGetStringLiteral(CORINFO_MODULE_HANDLE module, unsigned me
     }
 }
 
+void MethodContext::recObjectToString(void* handle, char16_t* buffer, int bufferSize, int length)
+{
+    if (ObjectToString == nullptr)
+        ObjectToString = new LightWeightMap<DLD, DD>();
+
+    DLD key;
+    ZeroMemory(&key, sizeof(key)); // Zero key including any struct padding
+    key.A = CastHandle(handle);
+    key.B = (DWORD)bufferSize;
+
+    DWORD strBuf = (DWORD)-1;
+    if (buffer != nullptr && length != -1)
+    {
+        int bufferRealSize = min(length, bufferSize) * sizeof(char16_t);
+        strBuf = (DWORD)ObjectToString->AddBuffer((unsigned char*)buffer, (unsigned int)bufferRealSize);
+    }
+
+    DD value;
+    value.A = (DWORD)length;
+    value.B = (DWORD)strBuf;
+
+    ObjectToString->Add(key, value);
+    DEBUG_REC(dmpObjectToString(key, value));
+}
+void MethodContext::dmpObjectToString(DLD key, DD value)
+{
+    printf("ObjectToString key hnd-%016llX bufSize-%u, len-%u", key.A, key.B, value.A);
+    ObjectToString->Unlock();
+}
+int MethodContext::repObjectToString(void* handle, char16_t* buffer, int bufferSize)
+{
+    if (ObjectToString == nullptr)
+    {
+        return -1;
+    }
+
+    DLD key;
+    ZeroMemory(&key, sizeof(key)); // Zero key including any struct padding
+    key.A = CastHandle(handle);
+    key.B = (DWORD)bufferSize;
+
+    int itemIndex = ObjectToString->GetIndex(key);
+    if (itemIndex < 0)
+    {
+        return -1;
+    }
+    else
+    {
+        DD value = ObjectToString->Get(key);
+        DEBUG_REP(dmpObjectToString(key, value));
+        int srcBufferLength = (int)value.A;
+        if (buffer != nullptr && srcBufferLength > 0)
+        {
+            char16_t* srcBuffer = (char16_t*)ObjectToString->GetBuffer(value.B);
+            Assert(srcBuffer != nullptr);
+            memcpy(buffer, srcBuffer, min(srcBufferLength, bufferSize) * sizeof(char16_t));
```

Comment:
Looks like this is missing null termination too

Reply:
doesn't it copy already zero terminated array so it's not needed?

Reply:
`srcBufferLength` is the recorded count of characters IIUC. And I guess it does not handle the empty string case

Reply:
and so is `srcBuffer` also recorded. So since empty string is handled in VMs is this still needed?

Reply:
Removed null-termination from API to match existing behavior of getStringLiteral